deployment/src/main/java/com/datastax/oss/quarkus/deployment/MetricsBuildStep.java

```diff
@@ -0,0 +1,161 @@
+        THROTTLING_DELAY,
+        Metadata.builder()
+            .withName(THROTTLING_DELAY.getPath())
+            .withDescription("How long requests are being throttled.")
+            .withType(MetricType.METERED)
+            .build());
+
+    supportedSessionMetrics.put(
+        THROTTLING_QUEUE_SIZE,
+        Metadata.builder()
+            .withName(THROTTLING_QUEUE_SIZE.getPath())
+            .withDescription("The size of the throttling queue.")
+            .withType(MetricType.GAUGE)
+            .build());
+
+    supportedSessionMetrics.put(
+        CQL_CLIENT_TIMEOUTS,
+        Metadata.builder()
+            .withName(CQL_CLIENT_TIMEOUTS.getPath())
+            .withDescription("The number of CQL requests that timed out.")
+            .withType(MetricType.COUNTER)
+            .build());
+
+    supportedSessionMetrics.put(
+        THROTTLING_ERRORS,
+        Metadata.builder()
+            .withName(THROTTLING_ERRORS.getPath())
+            .withDescription(
+                "The number of times a request was rejected with a RequestThrottlingException.")
+            .withType(MetricType.COUNTER)
+            .build());
+
+    supportedSessionMetrics.put(
+        CQL_PREPARED_CACHE_SIZE,
+        Metadata.builder()
+            .withName(CQL_PREPARED_CACHE_SIZE.getPath())
+            .withDescription("The size of the driver-side cache of CQL prepared statements.")
+            .withType(MetricType.GAUGE)
+            .build());
+
+    if (buildTimeConfig.metricsEnabled && capabilities.isCapabilityPresent(Capabilities.METRICS)) {
+      produceMetrics(metrics, supportedSessionMetrics, buildTimeConfig.metricsSessionEnabled);
+    }
+  }
+
+  private void produceMetrics(
+      BuildProducer<MetricBuildItem> metrics,
+      Map<SessionMetric, Metadata> supportedSessionMetrics,
+      Optional<List<String>> enabledMetrics) {
+    if (enabledMetrics.isPresent()) {
+      enabledMetrics
+          .get()
+          .forEach(
+              metricName ->
+                  supportedSessionMetrics.entrySet().stream()
+                      .filter((entry) -> entry.getValue().getName().equals(metricName))
+                      .forEach(entry -> produceMetric(entry.getKey(), entry.getValue(), metrics)));
+    }
+  }
+
+  private void produceMetric(
+      SessionMetric sessionMetric, Metadata metadata, BuildProducer<MetricBuildItem> metrics) {
+    MetricType typeRaw = metadata.getTypeRaw();
+    if (typeRaw.equals(MetricType.GAUGE)) {
+      metrics.produce(
+          new MetricBuildItem(metadata, new CassandraGauge(sessionMetric), true, CONFIG_ROOT));
+    } else if (typeRaw.equals(MetricType.METERED)) {
+      metrics.produce(
+          new MetricBuildItem(metadata, new CassandraMetered(sessionMetric), true, CONFIG_ROOT));
+    } else if (typeRaw.equals(MetricType.COUNTER)) {
+      metrics.produce(
+          new MetricBuildItem(metadata, new CassandraCounter(sessionMetric), true, CONFIG_ROOT));
+    }
```

Comment:
Don't need equals() here; you're comparing enum vals here so == is fine.  More directly: you're taking action based on an enum value which seems to cry out for a switch statement instead.